proposals/0297-relax-invalid-nonce.md

```diff
@@ -0,0 +1,81 @@
+---
+simd: '0297'
+title: Relax Invalid Nonced Transactions Constraint
+authors: Tao Zhu (Anza)
+category: Standard
+type: Core
+status: Review
+created: 2025-06-05
+feature:
+---
+
+## Summary
+
+This proposal relaxes the handling of invalid durable nonce transactions during
+block replay. Instead of rejecting the entire block when an invalid nonce is
+encountered, the transaction should be marked as failed, skipped for state
+modifications, but still committed to the block without charging a transaction
+fee.
+
+## Motivation
+
+The current consensus behavior specifies that invalid durable nonce transactions
+should result in block failure during replay. This behavior hinders forward
+compatibility with asynchronous block execution. The goal is to align invalid
+nonce handling with the treatment of other soft transaction failures (e.g.,
+relax fee payer, relax ALT, etc).
+
+## New Terminology
+
+None
+
+## Detailed Design
+
+### Current Behavior
+
+A transaction using a durable nonce fails block replay (causing the entire
+block to be rejected) if any of the following occurs:
+
+- The nonce account is not a statically included account.
+- The nonce account does not exist.
+- The nonce account is not properly initialized.
+- The stored nonce does not match the transaction's recent blockhash.
+- The transaction fails to advance the nonce.
+
+
+### Proposed Change
+
+Update replay logic to treat invalid nonce transactions — those failing the
+checks above — as non-state-modifying and non-fee-charging failures, as follows:
+
+- The transaction is not executed.
+- The transaction is not included in status cache.
+- The transaction’s non-execution CU cost (i.e., transaction's static CUs, plus
+  CUs for loading accounts) still applies to the block limit.
```

Comment:
We should clarify how this applies. Loading accounts is just the actually loaded accounts (fee-payer + nonce) or requested amount of data?

Reply:
Previous comments on this line said it is not the "requested" account bytes. The main point is "non-execution cost". How about just adding word "actual" to clarification: 
```suggestion
  actual CUs for loading accounts) still applies to the block limit.
```